networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/discovery/network/DiscoveryPeer.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.eth2.discovery.network;
+
+import static org.ethereum.beacon.discovery.schema.EnrField.IP_V4;
+import static org.ethereum.beacon.discovery.schema.EnrField.UDP_V4;
+
+import io.libp2p.core.PeerId;
+import io.libp2p.core.crypto.PubKey;
+import io.libp2p.crypto.keys.Secp256k1Kt;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.ethereum.beacon.discovery.schema.EnrFieldV4;
+import org.ethereum.beacon.discovery.schema.NodeRecord;
+import org.ethereum.beacon.discovery.schema.NodeRecordFactory;
+
+public class DiscoveryPeer {
+
+  static Logger logger = LogManager.getLogger();
+
+  private final Integer udpPort;
+  private final InetAddress address;
+  private final NodeRecord nodeRecord;
+
+  private DiscoveryPeer(
+      final NodeRecord nodeRecord, final InetAddress address, final Integer udpPort) {
+    this.address = address;
+    this.udpPort = udpPort;
+    this.nodeRecord = nodeRecord;
+  }
+
+  public Bytes getNodeId() {
```

Comment:
Would be really nice if we could condense `getNodeId`, `getPubKey`, and `getNodeIdString` into a single method: `NodeId getId()` (see [`NodeId`](https://github.com/PegaSysEng/artemis/blob/1b30ae4c3a9931299d1777fb0986a38e2405bbab/networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/peer/NodeId.java#L18)).